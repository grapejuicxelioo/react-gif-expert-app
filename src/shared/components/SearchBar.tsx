interface Props {
    placeHolder?: string,
}

export function SearchBar({ placeHolder }: Props) {
    return (
        <div className="search-container">
            <input type="text" placeholder={placeHolder}></input>
            <button>buscar</button>
        </div>
    )
}