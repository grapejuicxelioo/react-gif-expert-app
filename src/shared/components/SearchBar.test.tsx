import { describe, test, expect, vi } from 'vitest'
import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import { SearchBar } from './SearchBar'

describe("SearchBar", () => {
    test("should render searchbar correctly", () => {
        const { container } = render(<SearchBar onQuery={() => {}} />);

        expect(container).toMatchSnapshot();
        expect(screen.getByRole('textbox')).toBeDefined();
        expect(screen.getByRole('button')).toBeDefined();
    })
    test("should call onQuery with the correct value after x-ms", async () => {
        const onQuery = vi.fn();
        render(<SearchBar onQuery={onQuery} />);

        const input = screen.getByRole('textbox');
        fireEvent.change(input, {target: {value: 'test'}});

        // await new Promise((resolve) => setTimeout(resolve, 1000));
        await waitFor(() => {
            expect(onQuery).toHaveBeenCalled();
            expect(onQuery).toHaveBeenCalledWith('test');
        })
    })
})