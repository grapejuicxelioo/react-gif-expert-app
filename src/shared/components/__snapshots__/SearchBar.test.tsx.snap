// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBar > should .. 1`] = `
<div>
  <div
    class="search-container"
  >
    <input
      placeholder="buscar"
      type="text"
      value=""
    />
    <button>
      buscar
    </button>
  </div>
</div>
`;

exports[`SearchBar > should render searchbar correctly 1`] = `
<div>
  <div
    class="search-container"
  >
    <input
      placeholder="buscar"
      type="text"
      value=""
    />
    <button>
      buscar
    </button>
  </div>
</div>
`;
